panic due to send on closed channel

Seen during chain-sync when the connection was closed unexpectedly (perhaps due to a timeout). An immediate reconnect (perhaps reusing the same pointer var for the `ouroboros.Connection` object) may be involved.

```
panic: send on closed channel

goroutine 112 [running]:
github.com/blinklabs-io/gouroboros/protocol.(*Protocol).SendMessage(...)
	/go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.33.1/protocol/protocol.go:135
github.com/blinklabs-io/gouroboros/protocol/chainsync.(*Client).syncLoop(0xc0000aa400)
	/go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.33.1/protocol/chainsync/client.go:155 +0x8f
created by github.com/blinklabs-io/gouroboros/protocol/chainsync.(*Client).Sync
	/go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.33.1/protocol/chainsync/client.go:142 +0x21c
```

--- a/protocol/chainsync/client.go
+++ b/protocol/chainsync/client.go
@@ -146,7 +146,10 @@ func (c *Client) Sync(intersectPoints []common.Point) error {
 func (c *Client) syncLoop() {
 	for {
 		// Wait for a block to be received
-		<-c.readyForNextBlockChan
+		if _, ok := <-c.readyForNextBlockChan; !ok {
+			// Channel is closed, which means we're shutting down
+			return
+		}
 		c.busyMutex.Lock()
 		// Request the next block
 		// In practice we already have multiple block requests pipelined

